src/core/rtkQuery/splitV3Api/userApi.ts

```diff
@@ -30,54 +30,44 @@ const userApi = splitV3Api.injectEndpoints({
       providesTags: ['Users'],
     }),
 
-    // Edit User. This replaces all values, except Plex and Password.
-    putUser: build.mutation<void, UserType>({
+    // Add a new user.
+    postUser: build.mutation<UserType, PostUserType>({
       query: ({ CommunitySites, ...user }) => ({
         url: 'User',
-        method: 'PUT',
-        body: { ...user, CommunitySites: map(pickBy(CommunitySites, identity), (_, key) => key) },
+        method: 'POST',
+        body: { ...user, CommunitySites: CommunitySites ? map(pickBy(CommunitySites, identity), (_, key) => key) : null } as ApiPostUserType,
       }),
+      transformResponse: (user: ApiUserType) => {
+        const { CommunitySites, ...tempUser } = user;
+        return { ...tempUser, CommunitySites: simplifyCommunitySites(CommunitySites) };
+      },
       invalidatesTags: ['Users'],
     }),
 
-    // Change the password for a user
-    postChangePassword: build.mutation<void, { Password: string, RevokeAPIKeys: boolean, userId: number, admin: boolean }>({
-      query: ({ admin, userId, ...body }) => ({
-        url: `User/${admin ? userId.toString() : 'Current'}/ChangePassword`,
-        body,
-        method: 'POST',
+    // Edit the current user or a user by id using a raw object to do a partial update.
+    putUser: build.mutation<UserType, { ID?: number | undefined } & PutUserType>({
```

Comment:
Using `ID` as the user id field to make it so the normal `UserType` can be used as the `PutUserType`.

Reply:
Defeats the purpose of having a `PutUserType`
Just change ID to optional in `UserType` and use that